src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -71,10 +71,11 @@ pub struct Workunit {
 }
 
 impl Workunit {
-  fn log_workunit_state(&self) {
-    let state = match self.state {
-      WorkunitState::Started { .. } => "Starting:",
-      WorkunitState::Completed { .. } => "Completed:",
+  fn log_workunit_state(&self, canceled: bool) {
+    let state = match (&self.state, canceled) {
+      (_, true) => "Canceled:",
```

Comment:
I don't know this part of the codebase very well. Does it make sense to have `WorkunitState::Canceled` instead?

Reply:
We'll eventually probably want to represent workunit failure as part of the state, but adding that variant to the type now means we have to modify code in the pathway to the streaming workunit handler right now. I think we still need to do a little bit of thinking about how we want to present failed workunits to plugins, and that can come as a follow-up commit to this one.